### ISSUE TITLE
Panic on parsing a subquery with a range containing "M"

### ISSUE THREAD

**Issue by longtomjr** [VictoriaMetrics/VictoriaMetrics#4120]
### Describe the bug

When I call a query with a subquery range containing a capital letter `M`, the application crashes with the following panic: `panic: BUG: cannot parse duration "1M": cannot parse duration "1M"`

### To Reproduce

1. Call VM with a query like: `max_over_time(barman_last_wal_time[1M])`
2. The call will fail, and VM panics.

### Version

`victoria-metrics-20230225-020349-tags-v1.88.0-0-g2d36dbcfa`
`image: victoriametrics/victoria-metrics:v1.88.0`

### Logs

```
victoriametrics  | panic: BUG: cannot parse duration "1M": cannot parse duration "1M"
victoriametrics  |
victoriametrics  | goroutine 124 [running]:
victoriametrics  | github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper.func1()
victoriametrics  | 	github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:232 +0x65
victoriametrics  | panic({0xad5320, 0xc0026d4450})
victoriametrics  | 	runtime/panic.go:890 +0x263
victoriametrics  | github.com/VictoriaMetrics/metricsql.(*DurationExpr).Duration(0xc0026d4180, 0x6b20f0?)
victoriametrics  | 	github.com/VictoriaMetrics/metricsql@v0.56.1/parser.go:1370 +0xd5
victoriametrics  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncWithMetricExpr(0x0, 0xc000000780, {0xc0000b2390, 0xd}, 0x413956?, {0xdb5880, 0xc0005761b0}, 0xc00024b590, 0x0, 0xc0026d4180)
victoriametrics  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:1014 +0xaf
victoriametrics  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncWithoutAt(0xb21de0?, 0xc000000780, {0xc0000b2390?, 0x6be2d6?}, 0xc000541bc0?, {0xdb5880, 0xc0005761b0}, 0xc0005fb0c0, 0x0)
victoriametrics  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:825 +0x4e5
victoriametrics  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFunc(0xc0026d4410?, 0xc0000acd50?, {0xc0000b2390?, 0xd?}, 0x4?, {0xdb5880?, 0xc0005761b0?}, 0xc0031d4a38?, 0x7f2fecc53e10?)
victoriametrics  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:762 +0x44b
victoriametrics  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalExprInternal(0x10000bbaa24?, 0xc000000780, {0xdb5880?, 0xc0005761b0?})
victoriametrics  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:276 +0x52b
victoriametrics  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalExpr(0x114be30?, 0x27?, {0xdb5880?, 0xc0005761b0?})
victoriametrics  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:226 +0x267
victoriametrics  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.Exec(0xc0025ffb00?, 0xc000000780, {0xc0000b2390, 0x27}, 0x1)
victoriametrics  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/exec.go:63 +0x247
victoriametrics  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/prometheus.QueryHandler(0x0, {0xc0005400c0?, 0xb7759c?, 0x114f020?}, {0xdb8810, 0xc0005400c0}, 0xc0025ffb00)
victoriametrics  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/prometheus/prometheus.go:766 +0xa9e
victoriametrics  | github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect.RequestHandler({0xdb8810, 0xc0005400c0}, 0xc0025ffb00)
victoriametrics  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/main.go:256 +0xdd2
victoriametrics  | main.requestHandler({0xdb8810?, 0xc0005400c0}, 0xc0025ffb00)
victoriametrics  | 	github.com/VictoriaMetrics/VictoriaMetrics/app/victoria-metrics/main.go:121 +0x268
victoriametrics  | github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper(0xc000088690, {0xdb8810?, 0xc0005400c0}, 0xc0025ffb00, 0xd14af8)
victoriametrics  | 	github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:337 +0xd03
victoriametrics  | github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.gzipHandler.func1({0xdb8ba0?, 0xc0008282a0?}, 0xc000256ad0?)
victoriametrics  | 	github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:200 +0x76
victoriametrics  | net/http.HandlerFunc.ServeHTTP(0x0?, {0xdb8ba0?, 0xc0008282a0?}, 0x6771d4?)
victoriametrics  | 	net/http/server.go:2122 +0x2f
victoriametrics  | net/http.serverHandler.ServeHTTP({0xdb7e38?}, {0xdb8ba0, 0xc0008282a0}, 0xc0025ffb00)
victoriametrics  | 	net/http/server.go:2936 +0x316
victoriametrics  | net/http.(*conn).serve(0xc0009d46c0, {0xdb8fa8, 0xc000538ea0})
victoriametrics  | 	net/http/server.go:1995 +0x612
victoriametrics  | created by net/http.(*Server).Serve
victoriametrics  | 	net/http/server.go:3089 +0x5ed
```

### Used command-line flags

```
      - "--promscrape.config=/etc/prometheus/prometheus.yml"
      - "--storageDataPath=/storage"
      - "--httpListenAddr=:8428"
      - "--vmalert.proxyURL=http://vmalert:8880"
      - "--retentionPeriod=3y"
```

### Additional information

Possibly related to: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3589

I will update now, and update the issue. I just wanted to post the information I have first.

**Comment by Haleygo** [comment]
It does panic in here when [parse duration.](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/b9ab07ced94bb7f81d40fb0e52c11383fe1b3578/vendor/github.com/VictoriaMetrics/metricsql/parser.go#L1370)
<img width="759" alt="image" src="https://user-images.githubusercontent.com/39937150/231403091-96404a16-e9ef-4b7a-a9e8-8d5601a7fb89.png">

Looks like vm will panic under some similar scenario, not sure it's the best way to tell user that they passed wrong parameter
![image](https://user-images.githubusercontent.com/39937150/231403969-5625349e-e2c4-43e5-ba44-566424a2e8c0.png)

**Comment by Haleygo** [comment]
@dmitryk-dk Should we recover from this metricsql panic and report this error to user?

**Comment by dmitryk-dk** [comment]
Hi @Haleygo ! I think we should not recover from this panic because of the comment in the MetricsQL and it is designed to panic. I think @valyala can help us with this question.

**Comment by valyala** [comment]
The panic should be fixed in the commit 4b43c91f8ceef512ce572bf8e334094329e84a49 . This commit will be included in the next release. In the mean time it is possible to build VictoriaMetrics from this commit according to [these docs](https://docs.victoriametrics.com/#how-to-build-from-sources) and verify whether this fixes the panic.

**Comment by valyala** [comment]
The panic on queries containing durations with capital `M` should be fixed starting from [v1.91.0](https://docs.victoriametrics.com/CHANGELOG.html). See [this example at VictoriaMetrics playground](https://play.victoriametrics.com/select/accounting/1/6a716b0f-38bc-4856-90ce-448fd713e3fe/prometheus/graph/#/?g0.expr=rate%28foo%5B5M%5D%29&g0.range_input=30m&g0.end_input=2023-05-19T00%3A28%3A59&g0.relative_time=last_30_minutes).

Closing the issue as fixed then.

**Comment by valyala** [comment]
The bugfix has been backported to [v1.87.6 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.87.6).
